addons/mail/static/src/models/message_view.js

```diff
@@ -21,24 +21,35 @@ registerModel({
         /**
          * @param {MouseEvent} ev
          */
-        onClick(ev) {
+        async onClick(ev) {
             if (ev.target.closest('.o_channel_redirect')) {
-                this.messaging.openProfile({
+                // avoid following dummy href
+                ev.preventDefault();
+                const channel = this.messaging.models['Thread'].insert({
                     id: Number(ev.target.dataset.oeId),
                     model: 'mail.channel',
                 });
-                // avoid following dummy href
+                if (!channel.isPinned) {
+                    await channel.join();
+                    channel.update({ isServerPinned: true });
+                }
+                channel.open();
+                return;
+            } else if (ev.target.closest('.o_mail_redirect')) {
                 ev.preventDefault();
+                this.messaging.openChat({
+                    partnerId: Number(ev.target.dataset.oeId)
+                });
```

Comment:
Question related to other comment; if open chat doesn't work, don't we want to still open profile?

This spec is really making things confusing and harder to access I feel.